### PR TITLE
Refactor AT for event edit to utilise multiple features

### PR DESCRIPTION
**Pull request by tyom**
- Use better language for feature files, while retaining parameterisation
- Split page into separate features which can be reused across different contexts (Details to enable testing key/value tables and Form to contain form-specific features)
- Add `World` for sharing state between tests.

screen shot 2017-10-04 at 11 36 08

### CODE REVIEW

File: test/acceptance/features/events/step_definitions/edit.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+const World = require('../../../common/world')
+
 defineSupportCode(({ Before, Then, When }) => {
   const Form = client.page.Form()
   const Event = client.page.Event()
   const EventList = client.page.EventList()
 
   Before(() => {
-    Form.state = {}
+    World.resetState()
```
Comment (feedmypixel) [inline test/acceptance/features/events/step_definitions/edit.js]: I wonder if we could just use [hooks.js](http://mucsi96.github.io/nightwatch-cucumber/#hooks) and on all tests reset `World.resetState()`. This would be something to think of in another PR as it may be overkill running it for all.

File: test/acceptance/features/events/step_definitions/edit.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
   })
 
-  // Whens
+  // When
```
Comment (yeahfro) [inline test/acceptance/features/events/step_definitions/edit.js]: do we really need this?

File: test/acceptance/features/details/step_definitions/details.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const { client } = require('nightwatch-cucumber')
+const { defineSupportCode } = require('cucumber')
+
+const World = require('../../../common/world')
+
+defineSupportCode(({ Then }) => {
+  const Details = client.page.Details()
+
+  // Then
```
Comment (yeahfro) [inline test/acceptance/features/details/step_definitions/details.js]: do we really need this?

File: test/acceptance/features/events/step_definitions/edit.js (hunk 2 above)
Comment (tyom) [reply test/acceptance/features/events/step_definitions/edit.js]: I added those to be able to create visual hooks for groups of Whens and Thens when looking through a longer file. I find it easier to scan as they serve as titles.
